### ISSUE TITLE
Zarf fails local 'docker' fallback with docker-over-ssh

### ISSUE THREAD

**Issue by mjnagel** [zarf-dev/zarf#2943]
### Environment
Device and OS: macOS, with Ubuntu over SSH
App version: 0.38.3
Kubernetes distro being used: N/A

### Steps to reproduce
1. Setup docker daemon on a remote box that you can SSH to (this [tutorial](https://blog.programster.org/use-remote-docker-host-with-ssh) may be helpful).
2. Set your docker host to be the remote docker-over-ssh. Note that this must be done via the `DOCKER_HOST` env due to [this issue](https://github.com/zarf-dev/zarf/issues/2674). For example: `export DOCKER_HOST=ssh://mjnagel@1.2.3.4`.
3. Test some docker commands, validate connection works (could also validate with k3d and other tooling that interacts with the docker daemon).
4. Attempt to build a zarf package that requires the local docker fallback.

### Expected result

Zarf is able to pull the image from "local" docker daemon (or fails to find the image).

### Actual Result

Zarf fails to connect, appearing to try and hit the docker endpoint with an `http` call.

### Visual Proof (screenshots, videos, text, etc)

```
 WARNING  Falling back to local 'docker', failed to find the manifest on a remote: GET
          https://index.docker.io/v2/mjnagel/test123/manifests/latest: MANIFEST_UNKNOWN: manifest
          unknown; unknown tag=latest
     ERROR:  failed to create package: error during connect: Get "http://mjnagel%401.2.3.4/v1.24/images/docker.io/mjnagel/test123:latest/json": dial tcp: lookup mjnagel@1.2.3.4: no such host
```

### Severity/Priority

Medium - no known workarounds for this outside of being directly on the box with the docker daemon.

### Additional Context

Unsure if this is a bug or just a known limitation.

**Comment by brandtkeller** [comment]
Reproduced this issue locally. Need to review standard behavior of pulling images from a remote docker daemon before determining if this is a constraint.

**Comment by brandtkeller** [comment]
I believe this to be a limitation - albeit maybe not widely known. 

All logic for pulling images in zarf from a registry happens with connectivity of the machine to the associated registries. The logic for falling back to the local docker daemon requires http access to the local docker socket. In using the docker-over-ssh functionality - this socket is not directly accessible unless exposed via tcp. 

The client code to date expects tcp connectivity - evaluating native ssh access may be out of zarf scope.

**Comment by brandtkeller** [comment]
#2674 references docker context fixes to support multiple local docker daemon instances. Zarf can support this model as the socket is located locally and the `cli.Get()` requests are resolvable. 

Closing as outside of Zarf scope. Please re-open the issue if you believe otherwise.

**Comment by mjnagel** [comment]
@brandtkeller is it possible this could/should be documented somewhere as a limitation. Tracking the issues with implementation, and this is probably an edge case... but tooling like `k3d` does support this so it may be helpful to explicitly call this out as unsupported.

**Comment by brandtkeller** [comment]
Can you help me understand which features of other tools that support similar behavior?

I want to distinguish between commands that can communicate with docker-over-ssh vs those that explicitly pull images from the daemon with docker-over-ssh.

**Comment by mjnagel** [comment]
@brandtkeller yeah good clarification. I haven't used a lot of tooling but k3d/docker cli both support this in different ways:
- k3d: primarily creating network/containers
- docker cli: pretty much anything supported (haven't hit any limitations) - so create/pull/build/etc

The one caveat I've hit with both of those is volume mounts - essentially the volume mounts specified client side actually get evaluated "server side" (wherever the docker daemon is actually running). Notably a command like `docker save` does work the "way you'd expect", I can specify an image present on my remote docker daemon, and save it on my local client. Without knowing the code too in-depth that feels similar-ish to what Zarf is doing here?

**Comment by brandtkeller** [comment]
I agree - `docker save` is an instance of this where we do see similar functionality - otherwise I think much of the other tooling is using standard docker client/daemon workflows as you've mentioned vs non-standard workflows of pulling to the client such as the case of `docker save`. 

I need to examine more of the zarf code myself but It would be good to start documenting how zarf conducts some of these operations. IE the use of `crane` in zarf instead of `docker` means that the fallback to docker is inherently different from the standard/expected workflow - making zarf very different from other tools interacting with docker-over-ssh.  

This could be a feature request - where the team could then prioritize supporting this perhaps?

**Comment by mjnagel** [comment]
Yeah I think that would be a good path forward. And to provide the "external priority" from my end, I was just using this for a while to workaround some architectural restrictions (do things locally on my arm64 machine, deploy on an amd64 machine). There's definitely other paths to doing that and I haven't actually found myself doing this as frequently of late.
